core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -2776,13 +2776,11 @@ public boolean isValidRememberMeToken(String userName, String token)
             throw new IllegalArgumentException("Filter value cannot be null");
         }
         if (value.contains(QUERY_FILTER_STRING_ANY)) {
-            // This is to support LDAP like queries, so this will provide support for only leading or trailing '*'
-            // filters. if the query has multiple '*' s the filter will ignore all.
-                if ((value.startsWith(QUERY_FILTER_STRING_ANY) && !value.substring(1).contains(QUERY_FILTER_STRING_ANY)) ||
-                        value.endsWith(QUERY_FILTER_STRING_ANY) && !value.substring(0, value.length() - 1).contains(
-                                QUERY_FILTER_STRING_ANY) && value.charAt(value.length()-2) != SQL_FILTER_CHAR_ESCAPE) {
-                    value = value.replace(QUERY_FILTER_STRING_ANY, SQL_FILTER_STRING_ANY);
-                }
+            // This is to support LDAP like queries.
+            if (!(value.length() == 2 && value.startsWith(QUERY_FILTER_STRING_ANY) && value.endsWith
+                    (QUERY_FILTER_STRING_ANY)) && value.charAt(value.length() - 2) != SQL_FILTER_CHAR_ESCAPE) {
```

Comment:
Can't we remove,
`value.charAt(value.length() - 2) != SQL_FILTER_CHAR_ESCAPE` part?

I suppose that's covered in the reg-ex